operator/src/main/java/org/bf2/operator/managers/IngressControllerManager.java

```diff
@@ -483,11 +497,15 @@ private void buildIngressController(String name, String domain,
             replicas = Math.max(existingReplicas, replicas);
         }
 
+        Map<String, String> annotations = new HashMap<>();
+        hardStopAfter.ifPresent(value -> annotations.put(HARD_STOP_AFTER_ANNOTATION, value));
```

Comment:
```suggestion
hardStopAfter.ifPresent(value -> builder
                .editOrNewMetadata()
                .withAnnotations(Map.of(HARD_STOP_AFTER_ANNOTATION, value)));
```

I'm not sure if calling `editOrNewMetadata` is safe here, the alternative would be capture the reference to the Metadata on line 505 and use that in the ifPresent lambda. Would avoid any fun with empty vs null

Reply:
`editOrNewMetadata` is the right approach here, but I'd suggest using `.addToAnnotations(key, val)` to add a single annotation instead, to ensure you preserve any annotations that may be applied previously.